src/test/java/org/elasticsearch/index/query/RangeQueryBuilderTest.java

```diff
@@ -0,0 +1,157 @@
+
+    @Override
+    protected RangeQueryBuilder createTestQueryBuilder() {
+        RangeQueryBuilder query = new RangeQueryBuilder(randomAsciiOfLength(8));
+        // switch between numeric and date ranges
+        if (randomBoolean()) {
+            if (randomBoolean()) {
+                query.from(randomIntBetween(1, 100));
+                query.to(randomIntBetween(101, 200));
+            } else {
+                query.from(0.0-randomDouble());
+                query.to(randomDouble());
+            }
+        } else {
+            query = new RangeQueryBuilder(DATE_FIELD_NAME);
+            query.from(new DateTime(System.currentTimeMillis() - randomIntBetween(0, 1000000)).toString());
+            query.to(new DateTime(System.currentTimeMillis() + randomIntBetween(0, 1000000)).toString());
+            if (randomBoolean()) {
+                query.timeZone(TIMEZONE_IDS.get(randomIntBetween(0, TIMEZONE_IDS.size()-1)));
+            }
+            if (randomBoolean()) {
+                query.format("yyyy-MM-dd'T'HH:mm:ss.SSSZZ");
+            }
+        }
+        query.includeLower(randomBoolean()).includeUpper(randomBoolean());
+        if (randomBoolean()) {
+            query.boost(2.0f / randomIntBetween(1, 20));
+        }
+        if (randomBoolean()) {
+            query.queryName(randomAsciiOfLength(8));
+        }
+
+        if (randomBoolean()) {
+            query.from(null);
+        }
+        if (randomBoolean()) {
+            query.to(null);
+        }
+        return query;
+    }
+
+    @Override
+    protected void assertLuceneQuery(RangeQueryBuilder queryBuilder, Query query, QueryParseContext context) throws IOException {
+        assertThat(query.getBoost(), is(queryBuilder.boost()));
+        if (!queryBuilder.fieldname().equals(DATE_FIELD_NAME)) {
+            assertThat(query, instanceOf(TermRangeQuery.class));
+            TermRangeQuery termRangeQuery = (TermRangeQuery) query;
+            assertThat(termRangeQuery.includesLower(), is(queryBuilder.includeLower()));
+            assertThat(termRangeQuery.includesUpper(), is(queryBuilder.includeUpper()));
+            assertThat(termRangeQuery.getLowerTerm(), is(BytesRefs.toBytesRef(queryBuilder.from())));
+            assertThat(termRangeQuery.getUpperTerm(), is(BytesRefs.toBytesRef(queryBuilder.to())));
+            assertThat(termRangeQuery.getField(), is(queryBuilder.fieldname()));
+        } else {
+            assertThat(query, instanceOf(LateParsingQuery.class));
+            String expectedFromDate = expectedDateString(queryBuilder.from(), queryBuilder, context);
+            String expectedToDate = expectedDateString(queryBuilder.to(), queryBuilder, context);
+            String expectedLowerBracket = queryBuilder.includeLower() ? "[" : "{";
+            String expectedUpperBracket = queryBuilder.includeUpper() ? "]" : "}";
+            String queryString = query.rewrite(null).toString();
```

Comment:
instead of rewriting and relying on toString, could we maybe create the query that we expect and leverage lucene's equals implementation? Or if its our own query add equals and hashcode to it?